examples/cli/README.md

```diff
@@ -1,22 +1,95 @@
-# XLI
+# CLI
 
-XLI is an demo XMTPv3 console client, which allows developers to send and receive messages via the command line. While potentially useful its primary purpose is to demonstrate how to make a client application using the Rust Apis.
+![Status](https://img.shields.io/badge/Project_status-Alpha-orange)
 
-## Running
+This is a demo XMTP v3-alpha console client (CLI) that you can use to send and receive messages via the command line. Specifically, you can use it to try out [double ratchet messaging](#double-ratchet-messaging) and [installation key bundles](#installation-key-bundles) enabled by XMTP v3-alpha.
 
-### Register accounts
+> **Important**  
+> This software is in **alpha** status and ready for you to start experimenting with. Expect frequent changes as we add features and iterate based on feedback.
 
-`./xli.sh --db user1.db3 register`
-`./xli.sh --db user2.db3 register`
+## Send a double ratchet message
 
-### Get wallet address
+Use the CLI to send a [double ratchet message](#double-ratchet-messaging) between test wallets using a local XMTP node and database running in Docker Desktop.
 
-`./xli.sh --db user2.db3 info`
+1. In the root directory of this repo, run:
 
-### Send message
+   ```bash
+   dev/up
+   ```
 
-`./xli.sh --db user1.db3 send <user2_address> "hello"`
+2. Change directories to `examples/cli`.
 
-### List conversations
+   ```bash
+   cd examples/cli
+   ```
 
-`./xli.sh --db user1.db3 list-conversations`
+3. Create a sender wallet account (user1). Create an [installation key bundle](#installation-key-bundles) and store it in the database. Grant the installation key bundle permission to message on behalf of the sender address. This will allow the CLI to message on behalf of the sender address.
+
+   ```bash
+   ./xli.sh --db user1.db3 register
+   ```
+
+3. Likewise, create a recipient wallet account (user2) and an installation key bundle.
+
+   ```bash
+   ./xli.sh --db user2.db3 register
+   ```
+
+4. Get the recipient's wallet address.
+
+   ```bash
+   ./xli.sh --db user2.db3 info
+   ```
+
+5. Send a message into the conversation. The message is sent using [one session between each installation](#installation-key-bundles) associated with the sender and recipient. The message is encrypted using a session key dedicated to this single message per the [double ratchet algorithm](#double-ratchet-messaging).
+
+   ```bash
+   ./xli.sh --db user1.db3 send <user2_address> "hello"
+   ```
+
+6. List conversations.
+
+   ```bash
+   ./xli.sh --db user1.db3 list-conversations
+   ```
+
+## Double ratchet messaging
```

Comment:
This part (to the end) could probably be in the top level README. Doesn't seem specific to the CLI example.